docs/user/docker_config.rst

```diff
@@ -0,0 +1,158 @@
+        environment:
+             - PYTHONUNBUFFERED=1
+
+      # --------------------------------------------------------------------------
+      # OCS Agents
+      # --------------------------------------------------------------------------
+      ocs-aggregator:
+        image: simonsobs/ocs-aggregator-agent:latest
+        hostname: ocs-docker
+        user: "9000"
+        volumes:
+          - ${OCS_CONFIG_DIR}:/config:ro
+          - "/data:/data"
+        depends_on:
+          - "sisock-crossbar"
+
+      ocs-influx-publisher:
+        image: simonsobs/ocs-influxdb-publisher-agent:latest
+        hostname: ocs-docker
+        volumes:
+          - ${OCS_CONFIG_DIR}:/config:ro
+
+      ocs-LSA99ZZ:
+        image: simonsobs/ocs-lakeshore372-agent:latest
+        hostname: grumpy-docker
+        network_mode: "host"
+        volumes:
+          - ${OCS_CONFIG_DIR}:/config:ro
+        command:
+          - "--instance-id=LSA99ZZ"
+          - "--site-hub=ws://10.10.10.2:8001/ws"
+          - "--site-http=http://10.10.10.2:8001/call"
+
+
+.. warning::
+
+    Bind mounts are a system unique property. This is especially true for ones
+    which use absolute paths. If they exist in any reference configuration
+    file, they will need to be updated for your system.
+
+Understanding what is going on in this configuration file is key to getting a
+system that is working smoothly. The Docker Compose reference_ explains the
+format of the file, for details on syntax you are encouraged to check the
+official documentation.
+
+In the remainder of this section we will go over our example. The first line
+defines the version of the docker-compose file format, which corresponds to the
+Docker Engine version you are running. You likely do not have to change this.
+
+Every block below ``services:`` defines a Docker container. Let's look at one
+example container configuration. This example does not represent something we
+would want to actually use, but contains configuration lines relevant to many
+other container configurations::
+
+  example-container-name:
+    image: simonsobs/example-docker-image:latest
+    restart: always
+    hostname: ocs-docker
+    user: "9000"
+    ports:
+      - "127.0.0.1:8001:8001" # expose for OCS
+    volumes:
+      - /data:/data:ro
+      - ./.crossbar:/app/.crossbar
+    environment:
+        MAX_POINTS: 1000
+        SQL_HOST: "database"
+        SQL_DB: "files"
+    depends_on:
+      - "sisock-crossbar"
+      - "database"
+
+The top line, ``example-container-name``, defines the name of the service to
+docker-compose. These must be unique. ``image`` defines the docker image used
+for the container. A container can be thought of as a copy of an image. The
+container is what actually runs when you startup your docker service.
+``restart`` allows you to define when a container can be automatically
+restarted, in this instance, always. ``hostname`` defines the hostname internal
```

Comment:
Does this mean whenever `docker-compose up` is run the container is restarted?

Reply:
No, a `docker-compose up` will only restart a container if a new image has been pulled for it, or if a container that the one in question "depends on" is restarted.

`restart: always` will make sure the container is started on boot is the major thing for me. Other options are 'on-failure' and 'unless-stopped'. Perhaps 'always' is a bit aggressive, but I haven't run into any instances where it's been a terrible thing.

https://docs.docker.com/compose/compose-file/compose-file-v2/#restart

Reply:
Note, this has been replaced in more recent version of the compose standard with `restart_policy` which has some more control over when they're restarted: https://docs.docker.com/compose/compose-file/#restart_policy